File: June_17/EmployRestService/src/main/java/com/java/rest/service/EmployService.java
package com.java.rest.service;

import java.sql.SQLException;
import java.util.List;

import javax.ws.rs.Consumes;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import com.java.rest.dao.EmployDao;
import com.java.rest.dao.EmployDaoImpl;
import com.java.rest.model.Employ;

@Path("/employService")
public class EmployService {

	private static EmployDao employDao;
	
	static {
		employDao = new EmployDaoImpl();
	}
	
	@GET
	@Path("/showEmploy")
	@Produces(MediaType.APPLICATION_JSON)
	public List<Employ> showEmployService() throws SQLException {
		return employDao.showEmployDao();
	}
	
	@GET
	@Path("/searchEmploy/{empno}")
	@Produces(MediaType.APPLICATION_JSON)
	public Employ searchEmployService(@PathParam("empno") int empno) throws SQLException {
		return employDao.searchEmployDao(empno);
	}
	
	@POST
	@Path("/addEmploy")
	@Produces(MediaType.APPLICATION_JSON)
	@Consumes(MediaType.APPLICATION_JSON)
	public String addEmploy(Employ employ) throws SQLException {
		return employDao.addEmployDao(employ);
	}
}

File: June27_EmployService/src/app/employ.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Employ } from './employ';

@Injectable({
  providedIn: 'root'
})
export class EmployService {

  constructor(private _http : HttpClient) { }

  addEmploy(employ : Employ) {
    return this._http.post("http://localhost:9000/addEmploy",employ);
  }

  showEmploy() : Observable<any> {
    return this._http.get("http://localhost:9000/showEmploy");
  }

  searchEmploy(empno : number) : Observable<any> {
    return this._http.get("http://localhost:9000/searchEmploy/"+empno);
  }
}
